app/services/search/query_builders/query_base.rb

```diff
@@ -0,0 +1,52 @@
+module Search
+  module QueryBuilders
+    class QueryBase
+      attr_accessor :params, :body
+
+      def as_hash
+        @body
+      end
+
+      private
+
+      def build_body
+        @body = ActiveSupport::HashWithIndifferentAccess.new
+        build_queries
+        add_sort
+        set_size
+      end
+
+      def add_sort
+        sort_key = @params[:sort_by] || self.class::DEFAULT_PARAMS[:sort_by]
+        sort_direction = @params[:sort_direction] || self.class::DEFAULT_PARAMS[:sort_direction]
+        @body[:sort] = {
+          sort_key => sort_direction
+        }
+      end
+
+      def set_size
+        # By default we will return 0 documents if size is not specified
+        @body[:size] = @params[:size] || self.class::DEFAULT_PARAMS[:size]
+      end
+
+      def query_keys_present?
+        self.class::QUERY_KEYS.detect { |key| @params[key].present? }
+      end
+
+      def query_conditions
+        self.class::QUERY_KEYS.map do |query_key|
+          next if @params[query_key].blank?
+
+          {
+            simple_query_string: {
+              query: "#{@params[query_key]}*",
```

Comment:
Is the thinking here that this is common enough to be re-factored and it can be overridden, if necessary, by a `QueryBuilder` that searches differently? In other words, we're treating this "type" of search as the default?

`Tags` search is just slightly different and was implemented before we did `QueryBuilders`, for example.

Reply:
Yeah, if we need special treatment for classes we can override it but for now I figured why not extract it since its the same in 3 different places.

Reply:
Makes sense! Code Climate has been complaining about it too so this was prompted either way.